.github/workflows/check-learning-path-links.yml

```diff
@@ -0,0 +1,79 @@
+name: 'Check Learning Path Links'
+on:
+  schedule: # Run once a month
+    - cron: '0 0 1 * *'
+  workflow_dispatch:
+
+permissions: {}
+  
+jobs:
+  check-learning-path-links:
+    name: 'Check Learning Path Links'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout head
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          fetch-depth: 0 # Fetch the entire repo for the below git commit graph operations
+          ref: main
+
+      - name: Get current SHA
+        id: get_new_sha
+        run: |
+          new_sha=$(git rev-parse HEAD)
+          echo "new_sha=$new_sha" >> $GITHUB_ENV
+
+      - name: Get previous update SHA
+        id: get_sha
+        run: |
+          cd .github
+          prev_sha=$(cat learning-path-sha.txt)
+          echo "prev_sha=$prev_sha" >> $GITHUB_ENV
+          cd ../
+
+      - name: Checkout head
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: true
+          ref: main
```

Comment:
We already have `main` checked out. I'm assuming this was in response to my comment here: https://github.com/dotnet/dotnet-monitor/pull/5904#discussion_r1470101252

Having `persist-credentials: true` on the first checkout is okay since we aren't calling code we don't own/trust, but we should at least have a comment explaining that this is intention. Here's an example from another workflow:
https://github.com/dotnet/dotnet-monitor/blob/31385e3dceb2084584324fd011c9678e49439e2d/.github/workflows/cleanup-releases.yml#L23